### PR TITLE
rework to more nom style code

### PR DESCRIPTION
**Pull request by Stargateur**

### CODE REVIEW

**Comment by duesee** [comment]
Thank you! Could you also run

```sh
$ cargo fmt --config imports_granularity=Crate,group_imports=StdExternalCrate
```

and force push?

I will have a closer look the day after tomorrow, but your changes look very good! Happy to see the awkward `one` function be removed 👍🏻

File: src/complete.rs [hunk 1]
```diff
@@ -87,20 +87,14 @@ pub fn LF<'a, E: ParseError<&'a str>>(input: &'a str) -> IResult<&'a str, char,
 ///         ; Do not use when defining mail
 ///         ;  headers and use with caution in
 ///         ;  other contexts.
+// code as equivalent avoid branching LWSP = *([CRLF] WSP)
 pub fn LWSP<'a, E: ParseError<&'a str>>(input: &'a str) -> IResult<&'a str, &str, E> {
-    recognize(many0(alt((recognize(WSP), recognize(tuple((CRLF, WSP)))))))(input)
+    recognize(many0_count(terminated(opt(CRLF), WSP)))(input)
```
Comment (Stargateur) [inline src/complete.rs:92]: this also avoid the allocation of `many0`

Comment (Stargateur) [reply src/complete.rs:92]: could be write `recognize(many0_count(alt((WSP, preceded(CRLF, WSP)))))(input)`

**Comment by duesee** [comment]
Thanks!

**Pull request merged**